src/CoreWCF.Http/tests/AsyncNetAdoptionTests.cs

```diff
@@ -20,19 +21,26 @@ public AsyncNetAdoptionTests(ITestOutputHelper output)
         }
 
         [Fact]
-        public void OneWayPatternTest()
+        public async Task OneWayPatternTest()
         {
+            var inputs = Enumerable.Range(0, 10).Select(x => x.ToString());
             var host = ServiceHelper.CreateWebHostBuilder<AsyncNetAdoptionOneWayServiceStartup>(_output).Build();
             using (host)
             {
                 host.Start();
+                Services.OneWayService.TestOutputHelper = _output;
                 var httpBinding = ClientHelper.GetBufferedModeBinding();
                 var factory = new System.ServiceModel.ChannelFactory<ClientContract.IOneWayContract>(httpBinding,
                     new System.ServiceModel.EndpointAddress(new Uri("http://localhost:8080/OneWayPatternTest/basichttp.svc")));
-                var channel = factory.CreateChannel();                              
-                Task task = channel.OneWay("Hello");
+                var channel = factory.CreateChannel();
+                Parallel.ForEach(inputs, s => channel.OneWay(s));
+                await Task.WhenAll(Services.OneWayService.Tasks);
+                foreach (string input in inputs)
```

Comment:
There is still a race condition here as it's possible not all the request will have added a Task to OneWayService.Tasks yet. You could get a concurrent modification exception if WhenAll tries to iterate the list of Tasks before the final Task has been added. There's the same problem in the following foreach. You might have 9 Tasks in the list of Tasks which have completed, and the 10th is still pending execution and hasn't put its string into the Inputs yet.  
There's an easy fix though. Instantiate a CountdownEvent with the number of calls you want to have happened. In each service call, decrement the CountdownEvent using Signal. In the test, wait with a timeout on the CountdownEvent.